pegasus/helpers/email_preference_helpers.rb

```diff
@@ -0,0 +1,47 @@
+require 'cdo/email_preference_constants'
+require 'cdo/email_validator'
+
+class EmailPreferenceHelper
+  def self.upsert(email:, opt_in:, ip_address:, source:, form_kind:)
+    email = email&.strip&.downcase
+
+    raise "Email is required" unless !email.nil? && !email.empty?
+    raise "IP Address is required" unless !ip_address.nil? && !ip_address.empty?
+    raise "Source is required" unless !source.nil? && !source.empty?
+    raise "Opt In is required" unless [true, false].include?(opt_in)
```

Comment:
I would either change this message to reflect the validation, which is validating that Opt In is `true` or `false`, or change the check to check that `opt_in` is present.

Reply:
I updated the comment to explain that testing for presence of a boolean argument requires checking whether it is either true or false.